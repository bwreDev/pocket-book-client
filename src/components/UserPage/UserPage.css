.delete_debt,
.delete_income {
  background-color: #225a2a;
  border-radius: 50%;
  padding: 5px;
  border: none;
  text-align: center;
  text-decoration: none;
  font-size: 14px;
  width: 25px;
  margin: 5px;
}

.delete_debt:hover,
.delete_income:hover {
  color: red;
  cursor: pointer;
}

h3 {
  text-decoration: underline;
}

.income-section {
  background-color: rgba(97, 223, 66, 0.7);
  margin: 5px;
  border-radius: 15px;
}

.debt-section {
  background-color: rgba(236, 71, 71, 0.7);
  margin: 5px;
  border-radius: 15px;
}

.savings-section {
  background-color: rgba(129, 176, 224, 0.747);
  margin: 5px;
  border-radius: 15px;
}

ul {
  list-style: none;
  margin: 0;
  padding: 0;
}

@media only screen and (min-width: 600px) {
  .income-section {
    display: inline-flex;
    flex-direction: column;
    font-size: 20px;
    margin: 15px;
    padding: 15px;
  }

  .debt-section {
    display: inline-flex;
    flex-direction: column;
    font-size: 20px;
    margin: 15px;
    padding: 15px;
  }

  .savings-section {
    display: inline-flex;
    flex-direction: column;
    font-size: 20px;
    margin: 15px;
    padding: 15px;
  }
}
