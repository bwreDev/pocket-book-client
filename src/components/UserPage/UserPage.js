import React, { Component } from 'react';
import Navbar from '../../components/Navbar/Navbar';
import IncomeItem from '../../components/IncomeItem/IncomeItem';
import DebtItem from '../../components/DebtItem/DebtItem';
import './UserPage.css';
import InputsContext from '../../contexts/InputsContext';
import PocketBookApiService from '../../services/pocket-book-api-service';

export default class UserPage extends Component {
  state = {
    start_date: new Date(),
  };

  static contextType = InputsContext;

  componentDidMount() {
    PocketBookApiService.getInputs().then((inputs) =>
      this.context.setInputs(inputs)
    );
  }

  handleDelete = (id) => {
    PocketBookApiService.deleteInput(id).then((res) => {
      this.componentDidMount();
    });
  };

  getTotal = (inputs) => {
    let total = 0;
    for (let i = 0; i < inputs.length; i++) {
      total = total + parseFloat(inputs[i].amount);
    }
    return total;
  };

  getSavings = (inputs) => {
    let savings = 0;
    for (let i = 0; i < inputs.length; i++) {
      if (inputs[i].title === 'income') {
        savings += parseFloat(inputs[i].amount);
      }
      if (inputs[i].title === 'debt') {
        savings -= parseFloat(inputs[i].amount);
      }
    }
    return savings;
  };

  render() {
    return (
      <>
        <Navbar />
        <header role='banner'>
          <h1>Pocket Book!</h1>
          <h2>Welcome Back!</h2>
        </header>
        <section className='income-section'>
          <h3>Incomes</h3>
          <ul>
            {this.context.inputs
              .filter((i) => i.title === 'income')
              .map((i) => (
                <li key={i.id}>
                  <IncomeItem input={i} />
                  <button
                    className='delete_income'
                    onClick={() => this.handleDelete(i.id)}
                  >
                    X
                  </button>
                </li>
              ))}
          </ul>
          <h3>
            Total income: $
            {this.getTotal(
              this.context.inputs.filter((i) => i.title === 'income')
            )}
          </h3>
        </section>
        <section className='debt-section'>
          <h3>Debts</h3>
          <ul>
            {this.context.inputs
              .filter((i) => i.title === 'debt')
              .map((i) => (
                <li key={i.id}>
                  <DebtItem input={i} />
                  <button
                    className='delete_debt'
                    onClick={() => this.handleDelete(i.id)}
                  >
                    X
                  </button>
                </li>
              ))}
          </ul>
          <h3>
            Total debt: $
            {this.getTotal(
              this.context.inputs.filter((i) => i.title === 'debt')
            )}
          </h3>
        </section>
        <section className='savings-section'>
          <h3>Savings: $ {this.getSavings(this.context.inputs)}</h3>
        </section>
      </>
    );
  }
}
