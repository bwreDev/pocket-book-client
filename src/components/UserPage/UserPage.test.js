import React from 'react';
import ReactDOM from 'react-dom';
import UserPage from './UserPage';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router-dom';

it('should render without error', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <MemoryRouter>
      <UserPage />
    </MemoryRouter>,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('renders the UI as expected', () => {
  const tree = renderer
    .create(
      <MemoryRouter>
        <UserPage />
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
