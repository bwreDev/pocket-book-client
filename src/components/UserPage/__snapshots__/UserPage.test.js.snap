// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the UI as expected 1`] = `
Array [
  <nav
    className="navbar"
    role="navigation"
  >
    <ul
      className="navbar-list"
    >
      <li>
        <a
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          href="/register"
          onClick={[Function]}
        >
          Register
        </a>
      </li>
      <li>
        <a
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </nav>,
  <header
    role="banner"
  >
    <h1>
      Pocket Book!
    </h1>
    <h2>
      Welcome Back!
    </h2>
  </header>,
  <section
    className="income-section"
  >
    <h3>
      Incomes
    </h3>
    <ul />
    <h3>
      Total income: $
      0
    </h3>
  </section>,
  <section
    className="debt-section"
  >
    <h3>
      Debts
    </h3>
    <ul />
    <h3>
      Total debt: $
      0
    </h3>
  </section>,
  <section
    className="savings-section"
  >
    <h3>
      Savings: $ 
      0
    </h3>
  </section>,
]
`;
